### PR TITLE
Feat/unyt

### LINKED ISSUE #3: add unyt
https://unyt.readthedocs.io/en/stable/ is a great package for handling units and will simplify our workflow

### PR DESCRIPTION
**Pull request by jennyfothergill**
I'm working on adding [unyt](https://unyt.readthedocs.io/en/stable/) to make units conversions more explicit. fixes #3 

**This would be a breaking change**--~users used to specify the density in g/cm^3, but with this change planckton will have "default units" and _unless specified_ all values are assumed to be in these units, so density would be assumed to be amu/nm^3. How should a breaking change be handled?~

This PR means user _must_ specify their units.

### CODE REVIEW

**Comment by jennyfothergill** [comment]
BTW below is how I have been handling PRs that change the environment. Is this the right way to do it? 

1. Tag the current state of the repo which needs the new environment:
    `git tag <YOURTAGNAME>`
2. Push the tag: 
    `git push upstream --tags`
3. Switch the tagged version of the repo in environment.yml:
    (e.g `- git+https://github.com/cmelab/planckton@<YOURTAGNAME>`
4. Build the docker container (e.g. in planckton root dir):
    `docker build -f containers/dockerfile_cpu .`
(5.) Try it out:
    `docker image ls`  # get the latest image id 
    `docker run -it <IMAGEID> bash` # open a bash shell in the container
6. If everything works, tag and push the image:
    `docker tag <IMAGEID> cmelab/planckton:<VERSION>`
    `docker push cmelab/planckton:<VERSION>`
7. Set CI to use the just pushed image version. (e.g. edit the "image" line in .github/workflows/main.yml)

**Comment by chrisjonesBSU** [comment]
To be honest, I still think that density should be specified by at the user level in g/cm^3.  It's a lot more intuitive, and makes it easier to compare the simulation results with other papers.  I don't think there were any issues being caused by using g/cm^3 that changing to amu/nm^3 necessarily fixes, and I believe we can still implement the unyt package while sticking with g/cm^3 as the input density and handling any needed conversions when calculating the box edge length for mBuild's `packing,py`.

The biggest issue with how conversions were handled before is that it wasn't very clear where and how the conversion was being done. It was just the density being multiplied by a single number that contained multiple conversion factors wrapped up into one.

I like the addition of unyt, but I'd suggest sticking with g/cm^3 for the input density.

**Comment by erjank** [comment]
Being able to specify density in terms of volume fractions or number
density ar also important considerations if someone is using a coarse or
toy model that doesn't have physical unit connections (yet).

**Comment by mikemhenry** [comment]
We should make Eric's comment an issue, I think right now since planckton isn't being used by anyone outside of our lab, I think the breaking change is okay, but users shouldn't be able allowed to specify the density they want without having the units, then the only breaking change is the requirement for units, not an assumption about what units are being used.

**Comment by jennyfothergill** [comment]
@mikemhenry @erjank Issue #22 is made

So are we ok with the "units required" update?

**Comment by mikemhenry** [comment]
This PR is being weird for me right now, but it doesn't look like any commits have been made since my comment, so what I don't want is this 

> so density would be assumed to be amu/nm^3

I'm saying that we don't make any assumptions about density (or other user provided quants) and require units to be explicit

File: planckton/init.py [hunk 1]
```diff
@@ -1,25 +1,78 @@
...
+    """
+    Convenience class for filling box and atomtyping
+
+    Parameters
+    ----------
+    compound : Compound or list of Compounds
+        Compound(s) to initialize in simulation
+    n_compounds : int or list of ints
+        Number(s) of compound(s) to initialize
+    density : float or unyt.unyt_quantity
```
Comment (mikemhenry) [inline planckton/init.py]: I think this should be only a `unyt.unyt_quantity` and error on float

Comment (jennyfothergill) [reply planckton/init.py]: you got it 👍

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -11,15 +11,14 @@ jobs:
...
     runs-on: ubuntu-latest
     container: 
-      image: cmelab/planckton_cpu:0.0.1
+        image: cmelab/planckton_cpu:0.1.0
 
     steps:
     - uses: actions/checkout@v2
  
     - name: Run pytest with coverage report
       shell: bash -l {0}
```
Comment (mikemhenry) [inline .github/workflows/main.yml]:
```suggestion

```
Try removing this shell line

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -11,15 +11,13 @@ jobs:
...
     container: 
-      image: cmelab/planckton_cpu:0.0.1
+        image: cmelab/planckton_cpu:0.1.4
 
     steps:
     - uses: actions/checkout@v2
  
     - name: Run pytest with coverage report
-      shell: bash -l {0}
       run: |
```
Comment (mikemhenry) [inline .github/workflows/main.yml]:
```suggestion
      run: conda run -n planckton python -m pytest --cov=./ --cov-report=xml -v
```

File: .github/workflows/main.yml [hunk 4]
```diff
@@ -11,15 +11,13 @@ jobs:
...
-          conda activate planckton
           python -m pytest --cov=./ --cov-report=xml
```
Comment (mikemhenry) [inline .github/workflows/main.yml]:
```suggestion
```

**Pull request merged**